src/shared/components/mutationTable/MutationTable.tsx

```diff
@@ -0,0 +1,433 @@
+    cosmicData?:ICosmicData;
+    oncoKbData?:IOncoKbData;
+    mrnaExprRankGeneticProfileId?:string;
+    discreteCNAGeneticProfileId?:string;
+    columns?:MutationTableColumnType[];
+    data:Mutation[][];
+    initialItemsPerPage?:number;
+    itemsLabel?:string;
+    itemsLabelPlural?:string;
+    initialSortColumn?:string;
+    initialSortDirection?:SortDirection
+}
+
+export enum MutationTableColumnType {
+    SAMPLE_ID,
+    TUMORS,
+    GENE,
+    PROTEIN_CHANGE,
+    CHROMOSOME,
+    START_POS,
+    END_POS,
+    REF_ALLELE,
+    VAR_ALLELE,
+    MUTATION_STATUS,
+    VALIDATION_STATUS,
+    MUTATION_TYPE,
+    CENTER,
+    TUMOR_ALLELE_FREQ,
+    NORMAL_ALLELE_FREQ,
+    MUTATION_ASSESSOR,
+    ANNOTATION,
+    COSMIC,
+    COPY_NUM,
+    MRNA_EXPR,
+    COHORT,
+    REF_READS_N,
+    VAR_READS_N,
+    REF_READS,
+    VAR_READS
+}
+
+type MutationTableColumn = Column<Mutation[]>&{order?:number};
+
+export class MutationTableComponent extends LazyMobXTable<Mutation[]> {
+}
+
+export function getSpanForDataField(data:Mutation[], dataField:string) {
+    let contents = getTextForDataField(data, dataField);
+    return (<span>{contents}</span>);
+}
+
+export function getTextForDataField(data:Mutation[], dataField:string) {
+    let text = "";
+    if (data.length > 0 && data[0].hasOwnProperty(dataField)) {
+        text = (data[0] as any)[dataField];
+    }
+    return text;
+}
+
+export function defaultFilter(data:Mutation[], dataField:string, filterStringUpper:string):boolean {
+    if (data.length > 0) {
+        return data.reduce((match:boolean, next:Mutation)=>{
+            const val = (next as any)[dataField];
+            if (val) {
+                return match || ((val.toUpperCase().indexOf(filterStringUpper) > -1));
+            } else {
+                return match;
+            }
+        }, false);
+    } else {
+        return false;
+    }
+}
+
+@observer
+export default class MutationTable<P extends IMutationTableProps> extends React.Component<P, {}> {
+    @observable protected _columns:{[columnEnum:number]:MutationTableColumn};
+
+    public static defaultProps = {
+        initialItemsPerPage: 50,
+        initialSortColumn: "Annotation",
+        initialSortDirection: "desc",
+        itemsLabel: "Mutation",
+        itemsLabelPlural: "Mutations"
+    };
+
+    constructor(props:P)
+    {
+        super(props);
+        this._columns = {};
+        this.generateColumns();
+    }
+
+    protected getSamples():string[] {
+        if (this.props.sampleIds) {
+            return this.props.sampleIds as string[];
+        }
+        else {
+            return [];
+        }
+    }
+
+    protected generateColumns() {
+        this._columns = {};
+
+        this._columns[MutationTableColumnType.SAMPLE_ID] = {
+            name: "Sample ID",
+            render: SampleColumnFormatter.renderFunction,
+            sortBy: SampleColumnFormatter.getTextValue,
+            visible: true,
+            order: 10
```

Comment:
`order` should not be defined here, move this (and other order values) to either `PatientViewMutationTable` or `ResultsViewMutationTable`